GitUp/Application/Document.m

```diff
@@ -1965,8 +1965,33 @@ - (IBAction)openInFinder:(id)sender {
 // reset all permissions for particular bundle identifier.
 // $ tccutil reset All co.gitup.mac-debug
 
-- (IBAction)openInTerminal:(id)sender {
-  NSString* script = [NSString stringWithFormat:@"tell application \"Terminal\" to do script \"cd \\\"%@\\\"\"", _repository.workingDirectoryPath];
+- (NSString *)scriptForTerminalAppName:(NSString *)name {
+  if ([name isEqualToString:GIViewController_TerminalTool_Terminal]) {
+    return [NSString stringWithFormat:@"tell application \"%@\" to do script \"cd \\\"%@\\\"\"", name, _repository.workingDirectoryPath];
+  }
+  if ([name isEqualToString:GIViewController_TerminalTool_iTerm]) {
+    return [NSString stringWithFormat:
+            @"""tell application \"%@\" \n"""
+            """tell current session of current window \n"""
+            """set command to \"cd \\\"%@\\\"\" \n"""
+            """write text command \n"""
+            """end tell \n"""
+            """end tell""", name, _repository.workingDirectoryPath];
```

Comment:
So I noticed that this fails if I don't already have iTerm open. We should also match Terminal behavior which is to open a new window, not using the existing.

Also, I think we should leave the 2 tabs in settings instead of having the 3. I can see how it could be useful later on, but I don't think there is so much in that advanced tab yet to warrant another tab.

Lastly, we should check for existence of these applications without blindly showing iTerm if it's not even installed.

If those 3 things are addressed, this looks pretty good to go. Thanks for the work!

Reply:
@lucasderraugh 
What is the best way to hide iTerm entry in Preferences window? I suppose that Shared User defaults controller can do it, right?